redbot/core/commands/commands.py

```diff
@@ -106,6 +107,37 @@ def parents(self):
             # We should expose anything which might be a bug in the converter
             raise exc
 
+    async def can_see(self, ctx: "Context"):
+        """Check if this command is visible in the given context.
+
+        In short, this tries to emulate the behaviour of the help formatter
+        to see if this command would be displayed when the help command is
+        invoked in the given context.
+
+        Parameters
+        ----------
+        ctx : `Context`
+            The invocation context to check with.
+
+        Returns
+        -------
+        bool
+            ``True`` if this command is visible in the given context.
+
+        """
+        for cmd in itertools.chain((self,), self.parents):
```

Comment:
Is this really worth importing itertools for? This could easily just be

`for cmd in (self, *self.parents):`

Reply:
Good point, it's not like `parents` is going to be a particular memory-hungry sequence.